### PR TITLE
VFB-4 Add components for Icons

### PR DESCRIPTION
**Pull request by daniel-conlan-softwire**
Implemented icon components for Flag, Phone, Truck, Congestion Charge Applies, Feet, and Speech Bubble. Each icon has the appropriate on-hover text titles, and have been tested using unit tests for each.

- [X] Make sure you've verified it works via `npm run dev`
- [X] Make sure you've linted via `npm run lint`
    - can run `npm run fix` to try and fix as any mistakes automatically as possible!
- [X] Make sure you've tested via `npm run test`

### CODE REVIEW

File: src/components/Icons/FeetIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import Icon from "@/components/Icons/Icon";
+import { faShoePrints } from "@fortawesome/free-solid-svg-icons";
+import styled from "styled-components";
+
+interface Props {
+    collectionPoint: string;
+}
+
+const StyledIcon = styled(Icon)`
+    background-color: red;
```
Comment (Var1377) [inline src/components/Icons/FeetIcon.tsx:12]: Why is there a background colour on an icon?

**Review by Var1377 — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -28,6 +26,10 @@
   },
   "dependencies": {
     "@cypress/code-coverage": "^3.10.7",
+    "@fortawesome/fontawesome-svg-core": "^6.4.0",
+    "@fortawesome/free-regular-svg-icons": "^6.4.0",
```
Comment (TomCusack-Softwire) [inline package.json:30]: haven't used fortawesome before, but do you use `regular-svg-icons` ever? it appears you only ever get icons from `solid`.

File: src/components/Icons/CongestionChargeAppliesIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import React from "react";
+import Icon from "@/components/Icons/Icon";
+import { faCopyright } from "@fortawesome/free-solid-svg-icons";
+
+const CongestionChargeAppliedIcon: React.FC = () => {
```
Comment (TomCusack-Softwire) [inline src/components/Icons/CongestionChargeAppliesIcon.tsx:5]: Can a propsless Functional Component have the type parameter `React.FC<{}>` to make it explicitly clear? 🙂

File: src/components/Icons/FeetIcon.tsx (hunk 1 above)
Comment (TomCusack-Softwire) [inline src/components/Icons/FeetIcon.tsx:12]: Can you have default colours for each of the icons as per the Jira ticket? eg Congestion Charge is red, Speech Bubble is blue, ...
(we can [and likely will] change the exact shades later, but I'd like to have _something_ in from the start!)

File: src/components/Icons/Icons.cy.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        cy.get("svg").find("title").should("have.text", "Flagged for attention");
+    });
+
+    it("Truck icon text is correct", () => {
+        cy.mount(<TruckIcon />);
+        cy.get("svg").find("title").should("have.text", "Delivery");
+    });
+
+    it("Speech Bubble icon text is correct", () => {
+        cy.mount(<SpeechBubbleIcon onHoverText={"Text On HoVeR"} />);
```
Comment (TomCusack-Softwire) [inline src/components/Icons/Icons.cy.tsx:46]: why `HoVeR`? 😆

File: src/components/Icons/Icons.cy.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    });
+
+    it("Feet icon text is correct", () => {
+        cy.mount(<FeetIcon collectionPoint={"Next Door"} />);
+        cy.get("svg").find("title").should("have.text", "Collection at Next Door");
+    });
+
+    it("Phone icon text is correct", () => {
+        cy.mount(<PhoneIcon />);
+        cy.get("svg").find("title").should("have.text", "Requires follow-up phone call");
```
Comment (TomCusack-Softwire) [inline src/components/Icons/Icons.cy.tsx:27]: loving these tests! 😊

**Review by TomCusack-Softwire — CHANGES_REQUESTED** [review]

**Comment by TomCusack-Softwire** [comment]
Can your PR title also start with the ticket number and be in 'imperative case' - e.g. `VFB-123 Add text for buttons`. It's a very minor change but really helps with consistency / keeping Git clean!

File: src/components/Icons/Icons.cy.tsx (hunk 4 above)
Comment (daniel-conlan-softwire) [reply src/components/Icons/Icons.cy.tsx:46]: Just double checking that the capitalisation doesn't get removed on mount

Comment (TomCusack-Softwire) [reply src/components/Icons/Icons.cy.tsx:46]: is this not checked by 'Text on hover' (the capital T)?

**Review by TomCusack-Softwire — COMMENTED** [review]
Looks good! Will approve when two remaining comments are resolved :)

**Review by TomCusack-Softwire — APPROVED** [review]
Looks good! :)

**Pull request merged**